.github/workflows/ci-app-tests.yml

```
@@ -38,14 +42,28 @@ jobs:
         with:
           python-version: ${{ matrix.python-version }}
 
-      # TODO: use replace_oldest_ver() instead
+      - name: basic setup
+        run: pip install -q -r .actions/requirements.txt
+
       - name: Set min. dependencies
-        if: matrix.requires == 'oldest'
+        if: ${{ matrix.requires == 'oldest' }}
+        run: python .actions/assistant.py replace_oldest_ver
+      - run: python -c "print('TORCH_URL=https://download.pytorch.org/whl/' + str('test/cpu/torch_test.html' if '${{matrix.release}}' == 'pre' else 'cpu/torch_stable.html'))" >> $GITHUB_ENV
```

I think str is not necessary here?

```suggestion
      - run: python -c "print('TORCH_URL=https://download.pytorch.org/whl/' + 'test/cpu/torch_test.html' if '${{matrix.release}}' == 'pre' else 'cpu/torch_stable.html')" >> $GITHUB_ENV
```

---

I had it there but it did not helo as + was evaluated sooner the if/else